crates/aptos-crypto/src/traits.rs

```diff
@@ -62,22 +62,31 @@ pub trait ValidCryptoMaterial:
     // The for<'a> exactly matches the assumption "deserializable from any lifetime".
     for<'a> TryFrom<&'a [u8], Error=CryptoMaterialError> + Serialize + DeserializeOwned
 {
+    /// Prefix for AIP-80 e.g. ed25519-priv
+    const AIP_80_PREFIX: &'static str;
+
     /// Convert the valid crypto material to bytes.
     fn to_bytes(&self) -> Vec<u8>;
 }
 
-/// An extension to to/from Strings for [`ValidCryptoMaterial`].
+/// An extension to/from Strings for [`ValidCryptoMaterial`].
 ///
 /// Relies on [`hex`] for string encoding / decoding.
 /// No required fields, provides a default implementation.
 pub trait ValidCryptoMaterialStringExt: ValidCryptoMaterial {
     /// When trying to convert from bytes, we simply decode the string into
     /// bytes before checking if we can convert.
     fn from_encoded_string(encoded_str: &str) -> std::result::Result<Self, CryptoMaterialError> {
+        let mut str = encoded_str;
+        // First strip the prefix
+        if Self::AIP_80_PREFIX.is_empty() && str.starts_with(Self::AIP_80_PREFIX) {
+            str = str.strip_prefix(Self::AIP_80_PREFIX).unwrap()
```

Comment:
Can we remove the if and just do `str = str.strip_prefix(Self::AIP_80_PREFIX).unwrap_or(str)`?

Reply:
I had the if because it would add more computation for every parsing even if it didn't have an AIP-80 prefix, but yes I could just completely get rid of it.

I need to see what happens when it tries to strip an empty prefix

Reply:
Right.

We do use this `strip_prefix().unwrap_or()` pattern below for `0x` so it should be correct.

Performance should not be a concern here: it's just CLI serializations, right? Or, I guess this logic will be invoked in other parts too... But then `strip_prefix()` should return instantly when the prefix is empty. I guess the `unwrap_or` may cause unnecessary cloning/copying...

IDK. Up to you.

Reply:
This is used unfortunately everywhere in the node, so I went with this optimization.

Normally, I would agree with you, strip_prefix would be best to go by itself.

For simplicity, I will just put back strip prefix, Rust is pretty fast in this case, should be fine.